test/controllers/companies_controller_test.rb

```diff
@@ -5,25 +5,46 @@ def setup
     @company = companies(:Furaha)
   end
 
-  def file
+  def logo
     @file ||= File.open(File.expand_path( '../../file/3sixd.png', __FILE__))
   end
 
-  def uploaded_file_object(klass, attribute, file, content_type = 'image/png')
-    filename = File.basename(file.path)
+  def intro_image
+    @file ||= File.open(File.expand_path( '../../file/intro_image.png', __FILE__))
+  end
+
+  def uploaded_logo(klass, attribute, logo, content_type = 'image/png')
+    filename = File.basename(logo.path)
+    klass_label = klass.to_s.underscore
+   
+    ActionDispatch::Http::UploadedFile.new(
+      tempfile: logo,
+      filename: filename,
+      head: %Q{Content-Disposition: form-data; name="#{klass_label}[#{attribute}]"; filename="#{filename}"},
+      content_type: content_type
+    )
+  end
+
+  def logo_uploading
+    company = Company.last
+    assert_equal(File.basename(logo.path), company.logo_identifier)
+  end
+
+  def uploaded_intro_image(klass, attribute, intro_image, content_type = 'image/png')
```

Comment:
@leonardk1 This method takes care of the file upload in tests, it is a carrierwave testing method which will upload the intro_image for testing purpose, 

the most important part is 
```
filename = File.basename(intro_image.path)
  klass_label = klass.to_s.underscore
    ActionDispatch::Http::UploadedFile.new(
      tempfile: intro_image,
      filename: filename,
      head: %Q{Content-Disposition: form-data; name="#{klass_label}[#{attribute}]"; filename="#{filename}"},
      content_type: content_type
    )
``` which is what makes sure the intro_image is being uploaded for test before you can do the assert for that file in other blocks